mesa/space.py

```diff
@@ -34,12 +37,44 @@ class Grid(object):
         get_cell_list_contents: Returns the contents of a list of cells ((x,y) tuples)
     '''
 
-    width = None
-    height = None
-    torus = False
-    grid = None
     default_val = lambda s: None
 
+
+    class GridOccupiedIter:
+        """
+        Return just the (un)occupied cells of the grid.
```

Comment:
The grid does not need to be passed in! Just call the function I added to the Grid class: occupied_iter()

That will create an iterator for you that will return either the occupied or the unoccupied cells, depending on the occupied flag.

On Mar 10, 2015, at 11:17 PM, Jackie Kazil <notifications@github.com> wrote:

> In mesa/space.py:
> 
> >      default_val = lambda s: None
> >  
> > +
> > +    class GridOccupiedIter:
> > +        """
> > +        Return just the (un)occupied cells of the grid.
> 
> @gcallah -- for someone reading this -- what form does the grid need to be passed in? Can you add that and/or an example grid to the doc string?
> 
> �
> Reply to this email directly or view it on GitHub.
>